pkg/ccl/changefeedccl/changefeed_processors.go

```diff
@@ -704,22 +715,164 @@ func newKVEventToRowConsumer(
 	)
 
 	return &kvEventToRowConsumer{
-		frontier: frontier,
-		encoder:  encoder,
-		sink:     sink,
-		cursor:   cursor,
-		rfCache:  rfCache,
-		details:  details,
-		knobs:    knobs,
+		frontier:             frontier,
+		encoder:              encoder,
+		sink:                 sink,
+		cursor:               cursor,
+		rfCache:              rfCache,
+		details:              details,
+		knobs:                knobs,
+		topicDescriptorCache: make(map[TopicIdentifier]TopicDescriptor),
+		topicNamer:           topicNamer,
 	}
 }
 
 type tableDescriptorTopic struct {
-	catalog.TableDescriptor
+	tableDesc           catalog.TableDescriptor
+	spec                jobspb.ChangefeedTargetSpecification
+	nameComponentsCache []string
+	identifierCache     TopicIdentifier
+}
+
+// GetNameComponents implements the TopicDescriptor interface
+func (tdt *tableDescriptorTopic) GetNameComponents() []string {
+	if len(tdt.nameComponentsCache) == 0 {
+		tdt.nameComponentsCache = []string{tdt.spec.StatementTimeName}
+	}
+	return tdt.nameComponentsCache
+}
+
+// GetTopicIdentifier implements the TopicDescriptor interface
+func (tdt *tableDescriptorTopic) GetTopicIdentifier() TopicIdentifier {
+	if tdt.identifierCache.TableID == 0 {
+		tdt.identifierCache = TopicIdentifier{
+			TableID: tdt.tableDesc.GetID(),
+		}
+	}
+	return tdt.identifierCache
+}
+
+// GetVersion implements the TopicDescriptor interface
+func (tdt *tableDescriptorTopic) GetVersion() descpb.DescriptorVersion {
+	return tdt.tableDesc.GetVersion()
+}
+
+// GetTargetSpecification implements the TopicDescriptor interface
+func (tdt *tableDescriptorTopic) GetTargetSpecification() jobspb.ChangefeedTargetSpecification {
+	return tdt.spec
 }
 
 var _ TopicDescriptor = &tableDescriptorTopic{}
 
+type columnFamilyTopic struct {
+	tableDesc           catalog.TableDescriptor
+	familyDesc          descpb.ColumnFamilyDescriptor
+	spec                jobspb.ChangefeedTargetSpecification
+	nameComponentsCache []string
+	identifierCache     TopicIdentifier
+}
+
+// GetNameComponents implements the TopicDescriptor interface
+func (cft *columnFamilyTopic) GetNameComponents() []string {
+	if len(cft.nameComponentsCache) == 0 {
+		cft.nameComponentsCache = []string{
```

Comment:
do we envision more than 2 components? Does it make sense to have this return table/family names?